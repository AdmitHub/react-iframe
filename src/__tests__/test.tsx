import React from "react"
import Iframe from "../iframe"
import renderer from "react-test-renderer"

test("Render default iframe", () => {
	const component = renderer.create(
		<Iframe position="relative" url="http://www.foobar.com"/>
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})

test("Render iframe with title", () => {
	const component = renderer.create(
		<Iframe position="relative"
						title="A foobared iframe"
						url="http://www.foobar.com"
						ariaHidden={true}
						ariaLabel="someLabel"
						ariaLabelledby="someId"
						allow="encrypted-media"
						sandbox="allow-scripts"
						allowFullScreen={false}
						className="myIframeClass"
						frameBorder={2}
						overflow="auto"
						loading="eager"
						onLoad={()=>{console.log("hello")}}
						onMouseOut={()=>{console.log("goodbye")}}
						onMouseOver={()=>{console.log("hi")}}
						display="inline-block"
						id="myIframeId"
						name="My Iframe name"
						scrolling="yes"
						styles={{background: "red"}}
						target="_self"
		/>
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})

