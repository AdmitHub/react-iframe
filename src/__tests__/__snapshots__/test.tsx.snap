// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default iframe 1`] = `
<iframe
  allowFullScreen={false}
  frameBorder={0}
  height="100%"
  scrolling="no"
  src="http://www.foobar.com"
  style={
    Object {
      "display": "block",
      "height": "100%",
      "overflow": "hidden",
      "position": "relative",
      "width": "100%",
    }
  }
  target="_parent"
  width="100%"
/>
`;

exports[`Render iframe with title 1`] = `
<iframe
  allow="encrypted-media"
  allowFullScreen={false}
  aria-hidden={true}
  aria-labelledby="someId"
  className="myIframeClass"
  frameBorder={2}
  height="100%"
  id="myIframeId"
  name="My Iframe name"
  onLoad={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  sandbox="allow-scripts"
  scrolling="yes"
  src="http://www.foobar.com"
  style={
    Object {
      "display": "inline-block",
      "height": "100%",
      "overflow": "auto",
      "position": "relative",
      "width": "100%",
    }
  }
  target="_self"
  title="A foobared iframe"
  width="100%"
/>
`;
